sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -186,6 +186,11 @@ trait DateTimeFormatterHelper {
       }
       throw QueryExecutionErrors.failToRecognizePatternInDateTimeFormatterError(pattern, e)
   }
+
+  protected def checkInvalidPattern(pattern: String): PartialFunction[Throwable, Nothing] = {
+    case e: IllegalArgumentException =>
+      throw QueryExecutionErrors.failToRecognizePatternError(pattern, e)
```

Comment:
what's the difference between this one and `QueryExecutionErrors.failToRecognizePatternInDateTimeFormatterError`?

Reply:
I just renamed `failToRecognizePatternInDateTimeFormatterError` as `failToRecognizePatternAfterUpgradeError`. WDYT?